packages/tooling/src/main/kotlin/elide/tooling/kotlin/KotlinCompiler.kt

```diff
@@ -295,6 +309,92 @@ public val kotlinc: Tool.CommandLineTool = Tool.describe(
 
   /** Factories for configuring and obtaining instances of [KotlinCompiler]. */
   public companion object {
+    // Logging.
+    @JvmStatic private val ktLogger by lazy { Logging.of(KotlinCompiler::class) }
+
+    @JvmStatic private fun debugLog(message: String) {
+      if (KOTLINC_DEBUG_LOGGING) {
+        System.err.println("[kotlinc:debug] $message")
+      }
+    }
+
+    init {
+      initializeKotlinCompilerPlugins()
+    }
```

Comment:
compiler extensions / plugins are loaded at static init time, which should force these steps to be sealed into the image-persisted heap